### ISSUE TITLE
Range until is not working

### ISSUE THREAD

**Issue by pablobaldez** [pinterest/ktlint#2219]
## Expected Behavior

run ktlintFormat in this piece of code and have no errors
```kotlin
fun columns(count: Int) = when {
        count in 1 ..< 10 -> {
            ..
        }
        count >= 11 {
            ..
        }
        else -> error("count must be greater than 0. Current is $count.")
    }
```

## Observed Behavior
Failure with no clear description
```
* What went wrong:
Execution failed for task ':core:design:runKtlintFormatOverMainSourceSet'.
> A failure occurred while executing org.jlleitschuh.gradle.ktlint.worker.KtLintWorkAction
   > KtLint failed to parse file: MY_FILE

```

## Steps to Reproduce
using an rangeUntil operator and try to format the code

## Your Environment

* Version of ktlint used: 0.50.0
* Relevant parts of the `.editorconfig` settings
* Name and version (or code for custom task) of integration used (Gradle plugin, Maven plugin, command line, custom Gradle task): ktlintFormat
* Version of Gradle used (if applicable): 8.3
* Operating System and version: iOS Ventura

**Comment by AnthonyFillionMaillet** [comment]
Got the same issue so I used `rangeUntil` fun directly until they find a solution.

```
count in 1.rangeUntil(10) -> {

}
```

And it seems like another issue has been raised early on but got closed for some reason https://github.com/pinterest/ktlint/issues/1809

**Comment by paul-dingemans** [comment]
> ```
> fun columns(count: Int) = when {
>         count in 1 ..< 10 -> {
>             ..
>         }
>         count >= 11 {
>             ..
>         }
>         else -> error("count must be greater than 0. Current is $count.")
>     }
> ```
> 
`count >= 11 {` should be `count >= 11 -> {`

After fixing this error, ktlint CLI reports following:
```
ktlint-0.50.0 --stdin
21:25:04.219 [main] INFO com.pinterest.ktlint.cli.internal.KtlintCommandLine - Enable default patterns [**/*.kt, **/*.kts]
fun columns(count: Int) = when {
    count in 1 ..< 10 -> {

    }
    count >= 11 -> {

    }
    else -> error("count must be greater than 0. Current is $count.")
}
<stdin>:1:27: A multiline expression should start on a new line (standard:multiline-expression-wrapping)
<stdin>:1:27: Newline expected before expression body (standard:function-signature)
<stdin>:2:16: Unexpected spacing around "..<" (standard:range-spacing)
<stdin>:3:1: Unexpected blank line(s) before "}" (standard:no-blank-line-before-rbrace)
<stdin>:3:1: First line in a method block should not be empty (standard:no-empty-first-line-in-method-block)
<stdin>:6:1: Unexpected blank line(s) before "}" (standard:no-blank-line-before-rbrace)
<stdin>:6:1: First line in a method block should not be empty (standard:no-empty-first-line-in-method-block)
```

**Comment by paul-dingemans** [comment]
The problem might be caused by the gradle plugin that you are using and should be reported that their project.

Ktlint CLI version 0.50 does process code above without the error you mention. Please try to reproduce that yourself as well with CLI. See https://pinterest.github.io/ktlint/0.50.0/install/cli/
